### DISCUSSION TITLE
How can I cleanup/modify transaction names in PHP/Symfony

### DISCUSSION THREAD

**Discussion by mkilmanas** [getsentry/sentry-php#1335] [Q&A] [answered]
I want to group together transactions which are essentially the same thing. E.g. 
- `GET https://api.mydomain.com/foos/798y12031u0`
- `GET https://api.mydomain.com/foos/92kd10axl829`
- `GET https://api.mydomain.com/foos/ar3w98059jjx`
should all be grouped into 
- `GET https://api.mydomain.com/foos/{fooID}`

As the grouping seems to be happening by transaction name, I seem to be missing access to the event before it is submitted. [Transactions are explicitly bypassing before_send processor](https://github.com/getsentry/sentry-php/blob/master/src/Client.php#L273-L280) in favour of the traces_sampler, which in turn runs very early and the name/tags there do not reflect in the final submission.

So how can I achieve this grouping then? I was able to do this easily in the JavaScript version - that is very [nicely documented](https://docs.sentry.io/platforms/javascript/enriching-events/transaction-name/)

**Comment by stayallive** [comment, marked as answer]
See: getsentry/sentry-symfony#649, it's being tracked already and we are working on it!

**Reply by mkilmanas** [reply]
Glad to hear!

Any backstory as to why the transactions are being excluded from `before_send` processing? That sounds like a perfect place to hook into for customisations (and compared to the rest that should not have a significant performance penalty; it's optional anyway)

**Reply by stayallive** [reply]
There is: https://develop.sentry.dev/sdk/performance/#interaction-with-beforesend-and-event-processors.
